### PR TITLE
test: exclude lines only containing end-of-buffer from actual screen

### PR DESCRIPTION
**Pull request by luukvbaal**
Here's an idea: modify `screen.lua` to exclude end-of-buffer lines from the actual screen result.
What this accomplishes:
* Much smaller test files.
* Reduced visual clutter when reviewing added/failed tests.
* No need to `screen:try_resize()` to try to achieve this when writing tests.
* Profit???

I think we can do this freely without compromising the effectiveness of any test?

Tested on single file with `sed -i '/^[[:space:]]*{0:~[[:space:]]*}|/d' test/functional/ui/sign_spec.lua
`

### CODE REVIEW

File: test/functional/ui/screen.lua [hunk 1]
```diff
@@ -1292,7 +1292,10 @@ function Screen:render(headers, attr_state, preview)
     for i = 1, height do
       local cursor = self._cursor.grid == igrid and self._cursor.row == i
       local prefix = (headers or preview) and "  " or ""
-      table.insert(rv, prefix..self:_row_repr(igrid, i, attr_state, cursor).."|")
+      local row_repr = prefix..self:_row_repr(igrid, i, attr_state, cursor).."|"
+      if not row_repr:match("{?%d*:?~%s*}?") then
```
Comment (zeertzjq) [inline test/functional/ui/screen.lua:1296]: I think it's better to check the actual grid instead of the string returned by `_row_repr()`.

**Comment by zeertzjq** [comment]
I have another idea: add a notation to indicate the repetition of a line. That will work with splits.

**Comment by luukvbaal** [comment]
> I have another idea: add a notation to indicate the repetition of a line. That will work even if there are multiple splits.

Yeah I like that, bfredl had the same idea on matrix. Not because of splits (could be achieved with repeated pattern too), but for reduced "magicness".

**Comment by justinmk** [comment]
https://github.com/neovim/neovim/pull/26490

**Pull request closed without merging**